units/provisioning_setup_host.go

```diff
@@ -833,3 +841,59 @@ func parseLsblkOutput(lsblkOutput string) ([]blockDevice, error) {
 
 	return devices.BlockDevices, nil
 }
+
+// GetGithubTokensForTask returns a read-only token for owner/repo associated with the task
+// and a read-only token for each module associated with the task.
+func GetGithubTokensForTask(ctx context.Context, taskId string) (string, []string, error) {
+	catcher := grip.NewBasicCatcher()
+
+	var projectOwner, projectRepo string
+	var modules map[string]*manifest.Module
+	t, err := task.FindOneId(taskId)
+	catcher.Add(err)
+	if err == nil && t != nil {
+		mfest, err := manifest.FindFromVersion(t.Version, t.Project, t.Revision, t.Requester)
+		catcher.Add(err)
+		if err == nil && mfest != nil {
+			modules = mfest.Modules
+		}
+		p, err := model.FindMergedProjectRef(t.Project, t.Version, false)
+		catcher.Add(err)
+		if err == nil && p != nil {
+			projectOwner = p.Owner
+			projectRepo = p.Repo
+		}
+	}
+	var githubAppToken string
+	var moduleTokens []string
+	settings, err := evergreen.GetConfig(ctx)
+	catcher.Add(err)
+	opts := &github.InstallationTokenOptions{
+		Permissions: &github.InstallationPermissions{
+			Contents: utility.ToStringPtr(thirdparty.GithubPermissionRead),
+		},
+	}
+
+	if projectOwner != "" && projectRepo != "" && err == nil {
```

Comment:
nit: for consistency with the rest of the function, could check `settings != nil`